File: DataProcessor.java
import java.util.concurrent.*;
import java.util.*;
import java.io.*;

class Task {
    int id;
    Task(int id) { this.id = id; }

    void process() throws InterruptedException {
        Thread.sleep(500);  // Simulate processing
        System.out.println("Processed Task: " + id);
    }
}

class Worker implements Runnable {
    private BlockingQueue<Task> queue;
    private List<String> results;

    Worker(BlockingQueue<Task> queue, List<String> results) {
        this.queue = queue;
        this.results = results;
    }

    public void run() {
        try {
            while (true) {
                Task task = queue.poll(1, TimeUnit.SECONDS);
                if (task == null) break;
                task.process();
                synchronized (results) {
                    results.add("Result of Task " + task.id);
                }
            }
        } catch (Exception e) {
            System.err.println("Worker error: " + e.getMessage());
        }
    }
}

public class Main {
    public static void main(String[] args) throws Exception {
        BlockingQueue<Task> queue = new LinkedBlockingQueue<>();
        List<String> results = Collections.synchronizedList(new ArrayList<>());
        ExecutorService executor = Executors.newFixedThreadPool(4);

        for (int i = 0; i < 10; i++) queue.add(new Task(i));

        for (int i = 0; i < 4; i++) executor.submit(new Worker(queue, results));

        executor.shutdown();
        executor.awaitTermination(10, TimeUnit.SECONDS);

        try (PrintWriter out = new PrintWriter("output.txt")) {
            for (String line : results) out.println(line);
        }
    }
}

File: data_processor.go
package main

import (
    "fmt"
    "log"
    "os"
    "sync"
    "time"
)

type Task struct {
    ID int
}

func (t Task) process() {
    time.Sleep(500 * time.Millisecond)
    fmt.Printf("Processed Task: %d\n", t.ID)
}

func worker(id int, tasks <-chan Task, results *[]string, wg *sync.WaitGroup, mu *sync.Mutex) {
    defer wg.Done()
    for task := range tasks {
        task.process()
        mu.Lock()
        *results = append(*results, fmt.Sprintf("Result of Task %d", task.ID))
        mu.Unlock()
    }
}

func main() {
    tasks := make(chan Task, 10)
    var results []string
    var wg sync.WaitGroup
    var mu sync.Mutex

    for i := 0; i < 10; i++ {
        tasks <- Task{ID: i}
    }
    close(tasks)

    for i := 0; i < 4; i++ {
        wg.Add(1)
        go worker(i, tasks, &results, &wg, &mu)
    }

    wg.Wait()

    file, err := os.Create("output.txt")
    if err != nil {
        log.Fatal(err)
    }
    defer file.Close()
    for _, res := range results {
        fmt.Fprintln(file, res)
    }
}